conda_build/environ.py

```diff
@@ -191,6 +191,9 @@ def get_dict(m=None, prefix=None):
 
     elif sys.platform.startswith('linux'):      # -------- Linux
         d['LD_RUN_PATH'] = prefix + '/lib'
+        if cc.bits == 32:
+            d['CFLAGS'] = ' '.join([os.getenv('CFLAGS', ''), d.get('CFLAGS', ''), '-m32'])
+            d['CXXFLAGS'] = ' '.join([os.getenv('CXXFLAGS', ''), d.get('CXXFLAGS', ''), '-m32'])
 
```

Comment:
I think this isn't right.  You're copying`CFLAGS` from the user's environment in all cases, whereas I think it should only be copied iff they listed `CFLAGS` in their recipe's `script_env` section.  If you just get the `CFLAGS` via `d.get()` and not `os.getenv()`, then you'll get the right behavior.